autonomy/chain/tx.py

```diff
@@ -100,288 +106,136 @@ def __init__(
+                    self.tx_hash = None
+                    self.tx_dict = self._reprice(self.tx_dict or {})  # type: ignore
+                    continue
+
+                self.tx_dict = None
+                self.tx_hash = None
+                logger.error(
+                    f"Unexpected error occured when interacting with chain: {e}; "
                     f"will retry in {self.sleep}..."
                 )
                 time.sleep(self.sleep)
-        raise ChainTimeoutError("Timed out when waiting for transaction to go through")
 
-    def process(
-        self,
-        event: str,
-        receipt: Dict,
-        contract: PublicId,
-    ) -> Dict:
-        """Process tx receipt."""
-        return registry_contracts.get_contract(contract).get_events(
-            ledger_api=self.ledger_api,
-            contract_address=ContractConfigs.get(contract.name).contracts[
-                self.chain_type
-            ],
-            receipt=receipt,
-            event=event,
+        raise ChainTimeoutError(
+            f"Failed to send transaction after {self.retries} retries"
         )
 
-    # TODO The methods below (_transact() and transact_and_verify()) have been written in
-    # view of a future refactor of this class to avoid dependency on ContractConfigs class.
-    # These methods haven been added to the class rather than rewrite existing transact()
-    # and build() to avoid breaking changes.
-    def _transact(
-        self,
-        build_tx_contract: Contract,
-        build_tx_contract_address: str,
-        build_tx_contract_method: Callable,
-        build_tx_contract_kwargs: Dict,
-        dry_run: bool = False,
-    ) -> Dict:
-        """Make a transaction and return a receipt"""
-        retries = 0
-        tx_dict = None
-        tx_digest = None
-        already_known = False
-        deadline = datetime.now().timestamp() + self.timeout
-
-        if not (
-            inspect.ismethod(build_tx_contract_method)
-            and (
-                build_tx_contract_method.__self__ is build_tx_contract
-                or build_tx_contract_method.__self__ is build_tx_contract.__class__
-            )
-        ):
-            raise ValueError(
-                f"{build_tx_contract_method!r} must be a method of {build_tx_contract!r}"
-            )
-
-        while retries < self.retries and deadline >= datetime.now().timestamp():
-            retries += 1
-            try:
-                if not already_known:
-                    tx_dict = tx_dict or build_tx_contract_method(
-                        ledger_api=self.ledger_api,
-                        contract_address=build_tx_contract_address,
-                        raise_on_try=True,
-                        **build_tx_contract_kwargs,
-                    )
-                    if tx_dict is None:
-                        raise TxBuildError("Got empty transaction")
+    def settle(self) -> "TxSettler":
```

Comment:
What happens if you call transact().settle()  and then settle() again ? Does the class support being reused? or do we need instantiate a new TxObject for each transaction?